### DISCUSSION TITLE
Supplying more than 1 argument to groupby property in FTAggregateParameters

### DISCUSSION THREAD

**Discussion by ashleighplummer** [danitseitlin/redis-modules-sdk-ts#89] [Q&A]
I have an aggregate query where my number of groupby arguments (_nargs_) is more than 1.

When I supply more than 1 argument I get the following error: 

```Error: Redisearch: ReplyError: Bad arguments for GROUPBY: Expected an argument, but none provided```

How can I overcome this problem - Can you supply an example of using more than 1 argument within the FTAggregateParameters groupby?

Here's some example code:

```
const response = await client.aggregate(
	"index",
	`*`,
	{
		groupby: {
			nargs: "2",
			property: "@field1 @field2"
		},
	}
);
```

**Comment by ashleighplummer** [comment]
redis-modules-sdk/lib/modules/redisearch.js

aggregate function....

Replaced: 

```
args.push(parameters.groupby.property);
```

With:

```
parameters.groupby.property.split(" ").forEach((property) => {
              args.push(property);
          });
```

Entire Groupby if statement.
```
if (parameters.groupby !== undefined) {
      args.push('GROUPBY');
      if (parameters.groupby.nargs !== undefined)
          args.push(parameters.groupby.nargs);
      if (parameters.groupby.property !== undefined) {
          parameters.groupby.property.split(" ").forEach((property) => {
              args.push(property);
          });
      }
    }
```

**Reply by danitseitlin** [reply]
@ashleighplummer Hi, Fantastic catch - thanks!
Would you like to open a PR with the code changes?

**Reply by ashleighplummer** [reply]
@danitseitlin Sorry for the delayed response. Sure, I will try and submit within the next 24 hours. 👍🏾
